Elbow Saver Modules/Tests/ExerciseTimerTests/ExerciseTimerUnitTests.swift

```diff
@@ -0,0 +1,253 @@
+import XCTest
+import ComposableArchitecture
+@testable import ExerciseTimer
+
+
+final class ExerciseTimerUnitTests: XCTestCase {
+    // NB:- Syntax is a bit unusual if unfamilar with ComposableArchitecture.
+    // TestStore handles actions and compares its mutated state against
+    // the expected mutation passed in via the trailing closure.
+    // Absence of a closure is an assertion that there should be no state changes.
+    
+    
+    func testStartSession() {
+        let scheduler = DispatchQueue.test
+        let store = TestStore(
+            initialState: ExerciseTimerState(),
+            reducer: exerciseTimerReducer,
+            environment: .init(mainQueue: scheduler.eraseToAnyScheduler())
+        )
+        
+        store.send(.sessionStarted) { $0.currentTimerState = .performingRep }
+        
+        // ComposableArchitecture requires all effects to finish/be handled in tests, so we have to stop the timer.
+        // TODO: dluo - see if there is a way to cancel the timer without needing to send this action so this can be a true unit test
+        store.send(.stopButtonTapped) { $0.currentTimerState = .stopped }
```

Comment:
Don't forget about `override func tearDown()`, can do things like this for cleanup after each test automatically.

What you'd do is have a stored property in the class:
`var store: TestStore!` 

And either create this in `override func setup()` (**IF** initialState can be assigned after initialization)

or still init the store within each test:
`store = TestStore(initialState:...`

Don't forget to nil out the variable in the teardown though, to prevent any chance of leaked state between unit tests.

(Would recommend doing this in the integration test suite too if necessary)

Reply:
Would this still work here? Since I am asserting different conditions depending on the test when I send the stop action

Reply:
Potentially still could, if you can force the store into a state where it'll be guaranteed either finished or stopped upon cleanup

Reply:
Also, there's some value in initializing in `setup` (if `initialState` is mutable) as it cuts out a good amount of boilerplate. Namely, you wouldn't need to pass reducer and environment in every single test

Reply:
Can you just directly set the `store.state` without having to send an action?

Reply:
I definitely see the value in that. But yeah, part of the TCA state guardrails is that you can't mutate state in a `Store`/`TestStore` unless you're doing it by sending it an action, so I wouldn't be able to just start with one `Store` 😕 Believe me I did try! Heh.

Reply:
Nope, it's totally private!

<img width="493" alt="Screen Shot 2021-09-24 at 17 21 19" src="https://user-images.githubusercontent.com/62192491/134751408-ff6b1d6f-5931-4a90-bf1e-f4e1807d994f.png">

Reply:
Lame, guess there's not many options then. I'll take a look with you live

Reply:
Go for it, just try to keep an eye out for ways to improve this. Like you said maybe reset timerstate without an action if it's possible